### PR TITLE
Spatial.py update imported libraries to pep8, Edited documentation to pep8, suggested function edit

### LINKED ISSUE #54: Spatial.py edit documentation and import libraries to pep8 guidelines; suggest code edit for normalized_diff & stack_raster_tifs
## Identify the Documentation Element That You Propose to Update
Edit the documentation to match pep8 guidelines and rearrange the import libraries to follow pep8 guidelines wherein more general libraries are listed first and more specific libraries after. 

Also delete comments that seem unnecessary, and add comments to suggest a change to normalized_diff function to make it less confusing 

## Please Identify Your Proposed Changes Below

**Documentation should begin with a space and follow with a capital letter (i.e ' # Comment here')**

**Importing should begin with the most general libraries first**

**Delete comments such as**
1.)  #ndvi[np.isnan(ndvi)] = 0 *: commented out in normalized_diff function*
2.) # Should this wrap around show instead of plotting as it does? *: commented in plot_rgb*
3.)  #ax.set(xticks=[], yticks=[]) *:commented out in the hist function*

**Add comments**
1.) For normalized_diff function to suggest a change ( # Could be confusing if one does not know that b2 is subtracted from b1. 
#An individual might assume the first input is subtracted from the second. 
#For example, it could be more commonly assumed : (b1-b2)/(b1+b2) ) 

2.) Add comment for stack_raster_tifs function to include .sort()

### PR DESCRIPTION
**Pull request by shwh2628**
Editted documentation to pep8 guidelines (# Comment here) and deleted unnecessary code that was commented out. 

Included suggestions for code (normalized_diff and stack_raster_tifs) 

Arranged imported libraries to pep8 guidelines. 

Fix #54

### CODE REVIEW

**Comment by lwasser** [comment]
hey @shwh2628  thank you for this PR. in the future please be sure to tag whomever you are working with when you submit a PR to get their attention. in my case @lwasser would work! Reviewing this now.

File: earthpy/spatial.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import contextlib
 import os
 import geopandas as gpd
```
Comment (lwasser) [inline earthpy/spatial.py]: following proper order numpy, and os and contextlib should go at the top. Then matplotlib.  Then the spatil stuff then earthpy. please adjust the order @shwh2628  to follow conventions! geopandas is definitely considered a 3rd party package!

File: earthpy/spatial.py [hunk 2]
```diff
@@ -43,6 +43,9 @@ def extent_to_json(ext_obj):
 
 # Calculate normalized difference between two arrays
 # Both arrays must be of the same size
+# This could be confusing if one does not know that b2 is subtracted from b1.
```
Comment (lwasser) [inline earthpy/spatial.py]: @shwh2628 it is not appropriate to place additional commentary into a file. the goal is to REMOVE comments from a file like this not add more. 

If you wanted to adjust the structure of the function, you should submit a PR suggesting the band order be swapped in the code. Please REMOVE this change. 

this wold be a change you would have a discussion about with the package maintainers and if you looked closely at the issue you would see that discussion 

[here](https://github.com/earthlab/earthpy/issues/2) as an issue!

File: earthpy/spatial.py [hunk 3]
```diff
@@ -61,12 +64,12 @@ def normalized_diff(b1, b2):
...
 
     n_diff = (b2 - b1) / (b2 + b1)
-    #ndvi[np.isnan(ndvi)] = 0
     n_diff = np.ma.masked_invalid(n_diff)
     return n_diff
 
 
-# TODO: include a no data value here if provided ...
+# TODO: include a no data value here if provided 
+# TODO: Include .sort() in the function
```
Comment (lwasser) [inline earthpy/spatial.py]: this is another example of making a suggestion that the developers should approve. In this case i would argue that sort should happen after glob and before being passed to this function. This function's intent is not to order lists but rather to create a stack. It assumes the user passes what it wants. Please remove this  line frm your PR. 

A todo should be a conversation with the developers that they
1. agree to do and then
2. ideally your pr would fix that issue. 
Here i disagree that this is a todo.

**Comment by shwh2628** [comment]
Hi @lwasser , I (think) I pushed another commit. 

Did it work??

**Comment by lwasser** [comment]
it sure did. this looks good @shwh2628 !! i will merge it later today once the checks pass (i know they will) .  i'll ping you when i merge it :)

**Comment by lwasser** [comment]
hi @shwh2628 all tests have passed! i will merge this now! Thank you so much for this PR and congratulations on your first contribution to earthpy!

**Pull request merged**